orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/ToggleablePauseTask.java

```diff
@@ -0,0 +1,62 @@
+package com.netflix.spinnaker.orca.pipeline.tasks;
+
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService;
+import com.netflix.spinnaker.orca.api.pipeline.RetryableTask;
+import com.netflix.spinnaker.orca.api.pipeline.TaskResult;
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
+import java.time.Duration;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * A task that just waits indefinitely based on a feature toggle specified in the `pauseToggle`
+ * property of the stage context. Useful for pausing stages as part of migrations, for instance.
+ */
+@Component
+public class ToggleablePauseTask implements RetryableTask {
+  private final Logger log = LoggerFactory.getLogger(getClass());
+  private DynamicConfigService dynamicConfigService;
+
+  @Autowired
+  public ToggleablePauseTask(DynamicConfigService dynamicConfigService) {
+    this.dynamicConfigService = dynamicConfigService;
+  }
+
+  @Override
+  @Nonnull
+  public TaskResult execute(@Nonnull final StageExecution stage) {
+    final String pauseToggle =
+        stage.getContext().containsKey("pauseToggle")
+            ? stage.getContext().get("pauseToggle").toString()
+            : null;
+
+    if (pauseToggle != null && dynamicConfigService.isEnabled(pauseToggle, false)) {
+      log.debug(
+          stage.getName()
+              + " stage currently paused based on "
+              + pauseToggle
+              + " toggle. Waiting...");
```

Comment:
This formatting was spotless, not me... 🤷‍♂️

Reply:
Using the logger string replacement functionality is preferential to string concatenation as you’re doing here. It makes reading and writing the log statement easier in code.

Reply:
Changing the stage name is an interesting way to address this. Executions have a “system notification” concept that could be used instead that (i think) gets displayed as a notice on the executions view if you want to use that instead. Its like a pipeline execution-level blesk.